### PR TITLE
Add GeoJson adapter & test

### LINKED ISSUE #18: Add GeoJSON adapter for SQLDelight database
# Overview

The `json` column in the `feature` table will hold GeoJson strings, specifically for _Features._ In our code, we would like to handle these json strings as objects instead, and using SQLDelight's [custom column types](https://cashapp.github.io/sqldelight/multiplatform_sqlite/types/#custom-column-types) we can have these strings automatically convert themselves to and from their string representations so that we can use them as objects in our code.

In this case, the `json` column should be updated to store values as [`Feature`](https://dellisd.github.io/spatial-k/geojson/#feature) objects from Spatial K.

# Task Description

1. Update the SQLDelight table definition to adapt the `json` colum to the `Feature` class from Spatial K.
2. Implement a corresponding `ColumnAdapter` in common code.
2. Write unit tests to test the adapter on the **JVM**

### PR DESCRIPTION
**Pull request by safa-howaid**
Resolves #18

### CODE REVIEW

File: common/src/commonMain/kotlin/com/cuhacking/atlas/db/GeoJsonAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.cuhacking.atlas.db
+
+import com.squareup.sqldelight.ColumnAdapter
+import io.github.dellisd.spatialk.geojson.Feature
+import io.github.dellisd.spatialk.geojson.Feature.Companion.toFeature
+import kotlinx.serialization.UnstableDefault
+
+@OptIn(UnstableDefault::class)
+val geoJsonAdapter = object : ColumnAdapter<Feature, String> {
```
Comment (dellisd) [inline common/src/commonMain/kotlin/com/cuhacking/atlas/db/GeoJsonAdapter.kt:9]:
```suggestion
object GeoJsonAdapter : ColumnAdapter<Feature, String> {
```

File: common/src/androidTest/kotlin/GeoJsonAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        val featureAsObject= Feature(point(-75.0, 45.0))
+        val featureAsJson = "{" +
+                "\"type\":\"Feature\"," +
+                "\"geometry\":" +
+                "{" +
+                "\"type\":\"Point\"," +
+                "\"coordinates\":[-75.0,45.0]}," +
+                "\"properties\":{}" +
+                "}"
```
Comment (dellisd) [inline common/src/androidTest/kotlin/GeoJsonAdapterTest.kt:20]:
```suggestion
        val featureAsJson = 
            """
            { "type": "Feature",
              "geometry": {
                  "type": "Point",
                  "coordinates": [-75.0, 45.0],
              },
              "properties": {}
             }
             """
```

File: common/build.gradle.kts [hunk 3]
```diff
@@ -82,6 +81,10 @@ kotlin {
         implementation(deps.ktor.androidDriver)
     }
 
+    sourceSets["androidTest"].dependencies {
+        implementation("org.jetbrains.kotlin:kotlin-test-junit")
```
Comment (dellisd) [inline common/build.gradle.kts:85]: Can this dependency be added to the dependencies in `buildSrc`?

File: common/src/androidTest/kotlin/GeoJsonAdapterTest.kt (hunk 2 above)
Comment (safa-howaid) [reply common/src/androidTest/kotlin/GeoJsonAdapterTest.kt:20]: Yeah, that will probably make it easier to read 😬

File: common/build.gradle.kts (hunk 3 above)
Comment (safa-howaid) [reply common/build.gradle.kts:85]: Sure! I'll put it under the Kotlin object 👍

**Pull request merged**